esx_service/vmdk_ops.py

```diff
@@ -777,31 +776,90 @@ def disk_attach(vmdk_path, vm):
         except vim.fault.VimFault as ex:
             msg=("Failed to add PVSCSI Controller: %s", ex.msg)
             return err(msg)
+            
         # Find the controller just added
         devices = vm.config.hardware.device
         pvsci = [d for d in devices
                  if type(d) == vim.ParaVirtualSCSIController and
                  d.key == controller_key]
-        pci_slot_number = get_controller_pci_slot(vm, pvsci,
+        pci_slot_number = get_controller_pci_slot(vm, pvsci[0],
                                                   offset_from_bus_number)
 
     # Find a slot on the controller, issue attach task and wait for completion
     if not disk_slot:
-        taken = set([dev.unitNumber
+        idx = 0
+           
+        while ((disk_slot is None) and (idx < len(pvsci))):
```

Comment:
Nit, could remove the check for disk_slot at line 789 and can change this like below and avoid the check on disk_slot. 

idx = 0
while ((disk_slot is None) and (idx < len(pvsci))):
......